codex/stores/cachestore.nim

```diff
@@ -164,14 +174,7 @@ method listBlocks*(
           return not isManifest
     )
   )
-
-  return success(
-    map[Cid, ?Cid](
-      iter,
-      proc(cid: Cid): Future[?Cid] {.async.} =
-        some(cid),
-    )
-  )
+  success(iter)
```

Comment:
Why don't we need the map anymore?

Reply:
The `map` was just converting the yielded value to an optional because advertiser expects an `Option[Cid]` to be yielded. `SafeAsyncIter` yields a `Result` and so it can be directly used by the advertiser, and so mapping is not needed anymore. Also, in the original version, mapping was happening after filtering, and if there were something unexpected happening while iterating over `self.cache.keys` it would result in an exception. With `SafeAsyncIter`, if there is any exception while iterating it will result in a failure, which would then be detected by the advertiser. Because by default `SafeAsyncIter` stops on failure, we maintain the same semantics as before where an exception would interrupt the flow.